ios/engine/KMEI/KeymanEngine/Classes/Queries/Queries.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  Queries.swift
+//  KeymanEngine
+//
+//  Created by Joshua Horton on 6/25/20.
+//  Copyright © 2020 SIL International. All rights reserved.
+//
+
+import Foundation
+
+// The core `Queries` definition provides common utility methods, handlers, etc across our API queries.
+class Queries {
+  public enum FetchError: Error {
+    case networkError(Error)
+    case noData
+    case parsingError(Error)
+
+    var localizedDescription: String {
+      switch self {
+        case .noData:
+          return "no data returned"
+        case .networkError(let baseError):
+          return "network error occurred - \(String(describing: baseError))"
+        case .parsingError(let baseError):
+          return "failure occurred when parsing results - \(String(describing: baseError))"
+      }
+    }
+  }
+
+  public struct IDCodingKey: CodingKey {
+    /* Required by CodingKey, though we don't particularly need these */
+    public var intValue: Int?
+
+    public init?(intValue: Int) {
+      self.init(stringValue: "\(intValue)")
+      self.intValue = intValue
+    }
+    /* End section */
+
+    var stringValue: String
+
+    init?(stringValue: String) {
+      self.stringValue = stringValue
+    }
+  }
+
+  typealias JSONQueryCompletionBlock<T: Decodable> = (_: T?, _: Queries.FetchError?) -> Void
+  typealias DataTaskCompletionBlock = (_: Data?, _: URLResponse?, _: Error?) -> Void
+
+  /**
+   * Returns a closure that wraps closures expecting a decodable type,
+   * performing standard error checking and decoding the query's results
+   * on behalf of the wrapped closure.
+   *
+   * Returns either the decoded result when successful or the error that
+   * occurred when attempting to run the query.
+   */
+  internal static func jsonDataTaskCompletionAdapter<T: Decodable>(resultType: T.Type, completionBlock: @escaping JSONQueryCompletionBlock<T>) -> DataTaskCompletionBlock {
```

Comment:
This function is a significant abstraction of [APILexicalModelRepository's language-specific query code](https://github.com/keymanapp/keyman/blob/cf3c050f816cdf8f21e79a8da74ccef9553ffd81/ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/APILexicalModelRepository.swift), noting its internal code and matching code from [APIKeyboardRepository](https://github.com/keymanapp/keyman/blob/cf3c050f816cdf8f21e79a8da74ccef9553ffd81/ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/APIKeyboardRepository.swift).  

There's a fair bit of WETness in the 'source' queries; this abstraction aims to reduce it _dramatically_ if/when they are converted to using this method.  (Granted, we're about to _kill_ `APIKeyboardRepository` in the near future.)

While I'd love to go ahead and convert them to use this abstraction right now, that's probably out of scope at this time.  There's a bit of entanglement that would make proper unit testing a bit difficult, although the actual code replacement should be quite simple.  At a minimum, we should probably complete the full download rework first - I think the web-based embedded keyboard search takes priority.  (That's also when we "kill" the old keyboard search code.)